### PR TITLE
Surface synchronous approach for view components

### LINKED ISSUE #8553: What about no Async Work?
What if I don't have any async work to perform?  How should I write my `InvokeAsync` method?

>Warning	CS1998	This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread

```csharp
namespace ViewComponentSample.ViewComponents
{
    public class PriorityListViewComponent : ViewComponent
    {
        public async Task&lt;IViewComponentResult&gt; InvokeAsync(
        int maxPriority, bool isDone)
        {
            var items = new List<Item>{ ... };
            return View(items);
        }
    }
}
```
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9d2902f4-b6fd-ba1c-6762-d7c26f5eb4b9
* Version Independent ID: b5715d72-5f10-bb93-1051-7f06dfa9cde8
* Content: [View components in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/view-components?view=aspnetcore-2.0#avoiding-magic-strings)
* Content Source: [aspnetcore/mvc/views/view-components.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/mvc/views/view-components.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #8553 

cc: @spottedmahn

### CODE REVIEW

**Comment by guardrex** [comment]
Note for the record that `PriorityListViewComponent` doesn't work. It also doesn't work with `[ViewComponent(Name = "PriorityList")]`. Therefore, it's specific for the method name in this case. Let me know if you feel that I need to surface this. It doesn't really jive with what the topic states earlier about specifying a VC.

File: aspnetcore/mvc/views/view-components.md [hunk 1]
```diff
@@ -239,6 +239,43 @@ Add a `using` statement to your Razor view file, and use the `nameof` operator:
...
+    {
+        <li>@item</li>
+    }
+</ul>
+```
+
+The view component is invoked in a Razor file (for example, *Views/Home/Index.cshtml*) with:
+
+```cshtml
+@await Component.InvokeAsync(nameof(PriorityList), new { maxPriority = 4, isDone = true })
```
Comment (pranavkm) [inline aspnetcore/mvc/views/view-components.md]: Would it make sense to show the VCTagHelper as an example here? It hides the fact it's going async

Comment (guardrex) [reply aspnetcore/mvc/views/view-components.md]: Yes ... I'll do that ... and I'll say either do it this way or that way.

File: aspnetcore/mvc/views/view-components.md [hunk 2]
```diff
@@ -104,13 +104,13 @@ Pascal-cased class and method parameters for Tag Helpers are translated into the
 </vc:[view-component-name]>
 ```
 
-Note: In order to use a View Component as a Tag Helper, you must register the assembly containing the View Component using the `@addTagHelper` directive. For example, if your View Component is in an assembly called "MyWebApp", add the following directive to the `_ViewImports.cshtml` file:
+Note: In order to use a View Component as a Tag Helper, you must register the assembly containing the View Component using the `@addTagHelper` directive. For example, if your view component is in an assembly called `MyWebApp`, add the following directive to the `_ViewImports.cshtml` file:
```
Comment (scottaddie) [inline aspnetcore/mvc/views/view-components.md]: - ~~In order~~
- ~~you must~~
- Let's be consistent all throughout with the casing of "view component".
- Italicize the file name.
- While you're in the file, make the **Invoking a view component as a Tag Helper** section display ONLY for 1.1+.

File: aspnetcore/mvc/views/view-components.md [hunk 3]
```diff
@@ -239,6 +239,61 @@ Add a `using` statement to your Razor view file, and use the `nameof` operator:
...
+    {
+        <li>@item</li>
+    }
+</ul>
+```
+
+The view component is invoked in a Razor file (for example, *Views/Home/Index.cshtml*) using one of the following approaches:
+
+* <xref:Microsoft.AspNetCore.Mvc.IViewComponentHelper>
+* [Tag Helper](xref:mvc/views/tag-helpers/intro)
```
Comment (scottaddie) [inline aspnetcore/mvc/views/view-components.md]: Make it clear that the Tag Helper approach only works in 1.1+.

File: aspnetcore/mvc/views/view-components.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/mvc/views/view-components.md]: > Let's be consistent all throughout with the casing of "view component".

That's what I was trying to do. It was lowercase almost everywhere.

I'll get these on the next commit. :+1:

Comment (guardrex) [reply aspnetcore/mvc/views/view-components.md]: I want to simplify the prep phrase + subordinate conjunction just like we did the other day on that other PR. Let's just go with "If ... blah blah blah" ... "for example" is overused and obvious.

**Comment by guardrex** [comment]
oh ... one more commit coming ... stand by ...

**Comment by guardrex** [comment]
op op op ... we have it. All good. I'll merge this when the build completes.

**Pull request merged**